packages/notification-center/src/api/api.service.ts

```diff
@@ -28,58 +28,94 @@ export class ApiService {
+    });
   }
 
   async getNotificationsList(page: number, query: IStoreQuery = {}): Promise<IMessage[]> {
-    return await this.httpClient.get(`/widgets/notifications/feed`, {
-      page,
-      ...query,
+    return await this.defer(async () => {
+      return await this.httpClient.get(`/widgets/notifications/feed`, {
+        page,
+        ...query,
+      });
     });
   }
 
   async initializeSession(appId: string, subscriberId: string, hmacHash = null) {
-    return await this.httpClient.post(`/widgets/session/initialize`, {
-      applicationIdentifier: appId,
-      subscriberId: subscriberId,
-      hmacHash,
+    return await this.defer(async () => {
+      return await this.httpClient.post(`/widgets/session/initialize`, {
+        applicationIdentifier: appId,
+        subscriberId: subscriberId,
+        hmacHash,
+      });
     });
   }
 
   async postUsageLog(name: string, payload: { [key: string]: string | boolean | undefined }) {
-    return await this.httpClient.post('/widgets/usage/log', {
-      name: `[Widget] - ${name}`,
-      payload,
+    return await this.defer(async () => {
+      return await this.httpClient.post('/widgets/usage/log', {
+        name: `[Widget] - ${name}`,
+        payload,
+      });
     });
   }
 
   async getUnseenCount(query: IStoreQuery = {}) {
-    return await this.httpClient.get('/widgets/notifications/unseen', query as unknown as IParamObject);
+    return await this.defer(async () => {
+      return await this.httpClient.get('/widgets/notifications/unseen', query as unknown as IParamObject);
+    });
   }
 
   async getOrganization() {
-    return this.httpClient.get('/widgets/organization');
+    return await this.defer(async () => {
+      return this.httpClient.get('/widgets/organization');
+    });
   }
 
   async getUserPreference(): Promise<IUserPreferenceSettings[]> {
-    return this.httpClient.get('/widgets/preferences');
+    return await this.defer<IUserPreferenceSettings[]>(async () => {
+      return this.httpClient.get('/widgets/preferences');
+    });
   }
 
   async updateSubscriberPreference(
     templateId: string,
     channelType: string,
     enabled: boolean
   ): Promise<IUserPreferenceSettings> {
-    return await this.httpClient.patch(`/widgets/preferences/${templateId}`, {
-      channel: { type: channelType, enabled },
+    return await this.defer<IUserPreferenceSettings>(async () => {
+      return await this.httpClient.patch(`/widgets/preferences/${templateId}`, {
+        channel: { type: channelType, enabled },
+      });
+    });
+  }
+
+  private defer<T>(callback): Promise<T> {
+    let i = 0;
+
+    return new Promise((resolve, reject) => {
+      const interval = setInterval(() => {
+        if (this.isAuthenticated) {
+          clearInterval(interval);
+
+          return resolve(callback());
+        }
+        i++;
+        if (i >= 10) {
+          return reject();
```

Comment:
Should we add an Error message here? what will happen if the user is not authenticated (did not register within 100 ms)?

In case the user is authorized with bad credentials isAuthenticated will be true and the API will return 401, right?

Reply:
Like this is problem already today I can add a message in the reject... But not sure we like it to keep going in eternity.